climada/hazard/centroids/centr.py

```diff
@@ -172,6 +177,40 @@ def from_base_grid(land=False, res_as=360, base_file=None):
             centroids = centroids.select(reg_id=land_reg_ids)
         return centroids
 
+    @staticmethod
+    def from_geodataframe(gdf):
+        """Create centroids instance from GeoDataFrame. All columns beside the
+        geometry (or geom) column will be translated using pd.Series.to_numpy,
+        so the Series dtype will be respected.
+
+        >>> gdf = geopandas.read_file('centroids.shp')
+        >>> gdf.region_id = gdf.region_id.astype(int)  # type coercion
+        >>> centroids = Centroids.from_geodataframe(gdf)
+
+        Parameters:
+            gdf (GeoDataFrame): Where the geometry column needs to consist of
+                point features.
+        """
+        centroids = Centroids()
+
+        centroids.geometry = gdf.geometry
+        centroids.lat = gdf.geometry.y.to_numpy()
+        centroids.lon = gdf.geometry.x.to_numpy()
+
+        for col in gdf.columns:
+            if 'geom' in col:  # matches both 'geom' and 'geometry' columns
+                continue
+            val = gdf[col].to_numpy()
+            setattr(centroids, col, val)
+
+        if centroids.on_land.size == 0:
+            try:
+                centroids.on_land = ~np.isnan(centroids.region_id)
```

Comment:
Consider whether `np.isfinite()` would be more appropriate than `~np.isnan()`

Reply:
I don't know how many people know that np.NaN is not finite - whereas the check for "not NaN" seems quite intuitive to me. I would leave it like this...

Reply:
But then any non-valid region_id that is not NaN will be used ... 

You can leave as is, I do not have a strong opinion.

Reply:
I'll leave it as-is; this sort of problem should be prevented once we start type hinting or strong typing anyways.